services/galley/src/Galley/Intra/BackendNotificationQueue.hs

```diff
@@ -0,0 +1,62 @@
+{-# LANGUAGE NumericUnderscores #-}
+
+module Galley.Intra.BackendNotificationQueue where
+
+import Control.Lens (view)
+import Control.Monad.Catch
+import Control.Retry
+import Data.Domain
+import Data.Qualified
+import Galley.Effects.BackendNotificationQueueAccess (BackendNotificationQueueAccess (..))
+import Galley.Env
+import Galley.Monad
+import Galley.Options
+import Imports
+import qualified Network.AMQP as Q
+import Polysemy
+import Polysemy.Input
+import qualified System.Logger.Class as Log
+import UnliftIO.Timeout (timeout)
+import Wire.API.Federation.BackendNotifications
+import Wire.API.Federation.Error
+
+interpretBackendNotificationQueueAccess ::
+  ( Member (Embed IO) r,
+    Member (Input Env) r
+  ) =>
+  Sem (BackendNotificationQueueAccess ': r) a ->
+  Sem r a
+interpretBackendNotificationQueueAccess = interpret $ \case
+  EnqueueNotification remote deliveryMode action -> do
+    embedApp $ enqueueNotification (tDomain remote) deliveryMode action
+
+enqueueNotification :: Domain -> Q.DeliveryMode -> FedQueueClient c () -> App (Either FederationError ())
+enqueueNotification remoteDomain deliveryMode action = do
+  mChanVar <- view rabbitmqChannel
+  ownDomain <- view (options . optSettings . setFederationDomain)
+  case mChanVar of
+    Nothing -> pure (Left FederationNotConfigured)
+    Just chanVar -> do
+      let policy = limitRetries 3 <> constantDelay 1_000_000
+          handlers =
+            skipAsyncExceptions
+              <> [logRetries (const $ pure True) logError]
+      Right <$> recovering policy handlers (const $ go ownDomain chanVar)
+  where
+    logError willRetry (SomeException e) status = do
+      Log.err $
```

Comment:
This is something that may well happen every now and then without being an error.  eg., if only one backend-pusher is running and needs to be restarted, or during startup.  Should the log level be `info`, and only `error` once the retry count has been exhausted?

Reply:
This can only happen when RabbitMQ is down or unreachable. So, this is definitely an error.